binaries/aot_model_compiler.cc

```diff
@@ -93,10 +101,14 @@ int main(int argc, char** argv) {
   CAFFE_ENFORCE(!FLAGS_model_name.empty(), c10::UsageMessage());
   CAFFE_ENFORCE(!FLAGS_model_version.empty(), c10::UsageMessage());
   CAFFE_ENFORCE(!FLAGS_input_dims.empty(), c10::UsageMessage());
+  const auto dims_size = split(';', FLAGS_input_dims).size();
   CAFFE_ENFORCE(
-      split(';', FLAGS_input_dims).size() ==
-          split(';', FLAGS_input_types).size(),
+      dims_size == split(';', FLAGS_input_types).size(),
       "Number of input_dims and input_types should be the same");
+  const auto mem_formats_size = split(';', FLAGS_input_memory_formats).size();
+  CAFFE_ENFORCE(
+      mem_formats_size == 0 || mem_formats_size == dims_size,
```

Comment:
Since we give a default value for FLAGS_input_memory_formats at line 35, `mem_formats_size == 0` will never be true. Did you mean to remove the default string in line 35 as we set default value in line 261, and this condition would also work?